packages/datetime/src/dateRangeInput.tsx

```diff
@@ -304,10 +342,16 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     private handleStartInputChange = (e: React.FormEvent<HTMLInputElement>) => {
         this.handleInputChange(e, DateRangeBoundary.START);
+        if (this.props.startInputProps != null) {
+            Utils.safeInvoke(this.props.startInputProps.onChange, e);
+        }
     }
 
     private handleEndInputChange = (e: React.FormEvent<HTMLInputElement>) => {
         this.handleInputChange(e, DateRangeBoundary.END);
+        if (this.props.endInputProps != null) {
```

Comment:
I prefer to not deal with `null` as props input in general; it feels like code cruft. Let's not make any special cases for it and expect users to deal with those cases on their own